README.md

```diff
@@ -11,15 +11,60 @@ https://hub.docker.com/r/stshriv/kube-agent/
 
-2. ci.sh handles calling get_version.py in a vsts release definition. Provide ci.sh with the same parameters as get_version.
+### version.sh
+*version.sh* handles collecting the build IDs from a specified build definition so we can tag our images using them. version.sh handles calling into get_version.py.
 
-3. render.sh is used to create a dockerfile for each version returned by get_version.py. dockerfile.template provides a baseline dockerfile for building images with available msftkube wheels.
+To run version.sh use a Shell++ task with the filepath set to ci/version.sh
+
+Parameters:
+
+| Short Command |    Full Command    |  Type  | Required |   Default Value   |           Description           |
+|---------------|--------------------|--------|----------|-------------------|---------------------------------|
+|      -a       | --account          | string | required |                   | VSTS account name               |
+|      -b       | --branch           | string |          |'refs/heads/master'| VSTS Build Branch               |
+|      -c       | --count            |  int   |          |        0          | The number of builds to return  |
+|      -d       | --definition       |  int   | required |                   | The build definition ID         |
+|      -e       | --environmentOutput| string |          |                   | Script output                   |
+|      -p       | --project          | string | required |                   | VSTS Project within the account |
+|      -s       | --buildstatus      | string |          |                   | build status filter             |
+|      -t       | --token            | string |          |     NoneType      | User token to access account    |
+|      -u       | --user             | string |          |     NoneType      | Username associated with token  |
+|      -v       | --version          | float  |          |       4.1         | VSTS REST API version           |
+|      -V       | --verbose          | boolean|          |      False        | Print verbose output            |
+
+### render.sh
+*render.sh* handles the creation of version specific dockerfiles by substituting placeholder text within dockerfile.template.
+
+Use Shell++ script with the filepath set to ci/render.sh and provide the following parameters.
+1. Versions (file): A file containing the versions from version.sh. This file is specified with the -e parameter above.
+2. Output (filename): render will append to a file the full filepath of all dockerfiles generated after performing template substitution. Each line will contain `<docker file location> <build id/version>` for use in the build step of continuous integration.
+3. The placeholder text to substitute. 
+
+### build.sh
```

Comment:
I'm assuming the reasons we do our own build is 1) multiple versions 2) setting the version dynamically. 
what if though you had a release variable for the msftkube version it could default to latest but if we wanted to set it manually we could. We only get one version but we can hotfix an oldvesion if needed. 
If set to latest your get-versions script would go and populate it otherwise it's it's just a pass through. That might let you at least be able to use the existing docker build task rather than your own build.sh and the username/password setting it entails.

Reply:
Your assumptions are correct. If we want to do multiple versions we need to run this script. And building multiple versions is faster than running several builds separately since the machine will cache most of the docker image layers. Since we don't know which versions are available it becomes hard to pull the wheel unless we query VSTS.

About hotfixing,  once we have an image taggeed there will be a time when pulling the latest successful builds will not update an older image if it is no longer A) less than 30 days old OR B) It is not one of the ten most recent builds. However, once we have the image, we can render our templated dockerfile with whatever version we want to hotfix and rebuild the image that way.